core/src/main/java/org/ldaptive/transport/MessageFunctional.java

```diff
@@ -1,21 +1,22 @@
 /* See LICENSE for licensing and NOTICE for copyright. */
 package org.ldaptive.transport;
 
+import org.ldaptive.AbstractFreezable;
 import org.ldaptive.Connection;
 import org.ldaptive.OperationHandle;
 import org.ldaptive.Request;
 import org.ldaptive.Result;
 
 /**
- * Base class for processing a message that is initialized with the request and connection.
+ * Base class for processing a message that is initialized with properties from the handle.
  *
  * @param  <Q> type of request
  * @param  <S> type of result
  *
  * @author  Middleware Services
  */
 // CheckStyle:AbstractClassName OFF
-public abstract class MessageFunctional<Q extends Request, S extends Result>
+public abstract class MessageFunctional<Q extends Request, S extends Result> extends AbstractFreezable
```

Comment:
You have a bit of a naming issue here with an abstract base class that doesn't begin with the typical `Abstract` name. May I suggest `AbstractMessageBasedResultHandler`.

Reply:
I'll take that under advisement for a future patch.